### PR TITLE
Fix an issue when adding a node then removing a node caused "Contains…

### PR DESCRIPTION
**Pull request by jmfallecker**
…" function for that node to thrown a NullReferenceException instead of returning "false"

### CODE REVIEW

**Comment by jmfallecker** [comment]
Well, the first commit should definitely go in.

I was going only off of current code functionality (what looks like should be that if a node with state REM is attempted to be removed, that it returns false.

This seems to break the Integration test.

Hope you're well, and nice implementation of this!

-Joe

**Comment by danielkylewood** [comment]
Sorry dude, just returned from vacation and saw this. Will have a look tomorrow.

File: src/ConcurrentLinkedList/ConcurrentLinkedList.cs [hunk 1]
```diff
@@ -122,15 +122,16 @@ private static bool HelpRemove(Node<T> node, T value, out T result)
                     previous.Next = successor;
                     current = successor;
                 }
+                else if (state == NodeState.REM)
```
Comment (danielkylewood) [inline src/ConcurrentLinkedList/ConcurrentLinkedList.cs:125]: I don't think moving this check here is right. It solves the immediate problem, but stops threads finding the nodes they are looking to remove (hence the failing tests). I think solution is to change the node value check to also check for null, as you did in first commit. 

So `else if (!current.Value.Equals(value))` => `else if (current.Value != null && !current.Value.Equals(value))` and move this check back to where it was. Your added test passes with this as well.

**Review by danielkylewood — CHANGES_REQUESTED** [review]

**Comment by danielkylewood** [comment]
I think with the one suggested change then all works fine. Your assumptions on expected behaviour are right, the exceptions were definitely an oversight on my part so thanks!

**Comment by danielkylewood** [comment]
Closing due to inactivity

**Pull request closed without merging**